packages/xen-api/src/index.js

```diff
@@ -653,31 +654,100 @@ export class Xapi extends EventEmitter {
   // Private
   // ===========================================================================
 
+  @synchronized()
+  async _updateUrlFromFallbackAdresses(method, args, startTime) {
+    debug('%s: will try to find a new master', this._humanId)
+    const hostAddresses = this._fallBackAddresses
+
+    // there can be a master only if there is more than 1 host
+    if (hostAddresses.length < 2) {
+      return false
+    }
+
+    // another call to _updateUrlFromFallbackAdresses has already connected us to the right host
+    try {
+      await this.transport('session.get_auth_user_name', [])
```

Comment:
I'm not sure I like this because it breaks non-session calls (like when opening a new session).

I would prefer for this function to call the expected method directly.

Reply:
If I use _call, then I won't have EHOSTUNREACH or HOST IS SLAVE ( they're caught , and they launch again _call). If I directly call `this._transport(method, args)` I will lose the error and timeout  handling of `_call`
Also, if the session is not openned, the call should fail with an error `SESSION_INVALID` that is hanlded here.
The last argument is that the `syncrhonized` part will incur a performance cost, and I would like to keep this part as small and contained and  as possible.

Reply:
I don't mean calling  `_call` instead of `transport`, I mean calling the RPC method `method` instead of `session.get_auth_user_name`.